synapse/config/key.py

```diff
@@ -402,7 +406,9 @@ def _perspectives_to_key_servers(config):
 }
 
 
-def _parse_key_servers(key_servers, federation_verify_certificates):
+def _parse_key_servers(
+    key_servers: List[Any], federation_verify_certificates: bool
```

Comment:
All we know is that it is a list, we then verify that the list is the proper form below, using JSON schema. The input is from YAML so could really be anything.

Reply:
Oh, hmmm I see what you're saying it is either a hard-coded value of the result of `_perspectives_to_key_servers`, so maybe we can do better here, yes!

Reply:
Ah, but we can't since we *extend* it with `_perspectives_to_key_servers`, so the original gunk could still just be weird values. We really have no idea what it is.

Reply:
Ah yeah good point.